### PR TITLE
add ProxyType

### PR DESCRIPTION
**Pull request by zjb0807**
Closes: #1075

### CODE REVIEW

File: runtime/mandala/src/lib.rs [hunk 1]
```diff
@@ -1250,11 +1250,113 @@ parameter_types! {
...
+				Call::CdpTreasury(..) |
+				Call::CdpEngine(..) |
+				Call::EmergencyShutdown(..) |
+				Call::Homa(..) |
+				Call::NomineesElection(..) |
+				Call::StakingPool(..) |
+				Call::PolkadotBridge(..) |
+				Call::HomaValidatorListModule(..) |
+				Call::Incentives(..) |
+				Call::NFT(..) |
```
Comment (xlc) [inline runtime/mandala/src/lib.rs:1323]:
```suggestion
```

File: runtime/mandala/src/lib.rs [hunk 2]
```diff
@@ -1250,11 +1250,113 @@ parameter_types! {
...
+				Call::AirDrop(..) |
+				Call::EVM(..) |
```
Comment (xlc) [inline runtime/mandala/src/lib.rs:1325]:
```suggestion
```

File: runtime/mandala/src/lib.rs [hunk 3]
```diff
@@ -1250,11 +1250,113 @@ parameter_types! {
...
+				c,
+				Call::Authority(..)
+					| Call::GeneralCouncil(..)
+					| Call::FinancialCouncil(..)
+					| Call::HomaCouncil(..)
+					| Call::TechnicalCommittee(..)
+					| Call::Treasury(..) | Call::Bounties(..)
+					| Call::Tips(..) | Call::Utility(..)
+			),
+			ProxyType::Staking => matches!(c, Call::Session(..) | Call::Utility(..)),
```
Comment (xlc) [inline runtime/mandala/src/lib.rs:1340]: also `CollatorSelection`

File: runtime/mandala/src/lib.rs [hunk 4]
```diff
@@ -1250,11 +1250,111 @@ parameter_types! {
...
+impl Default for ProxyType {
+	fn default() -> Self {
+		Self::Any
+	}
+}
+impl InstanceFilter<Call> for ProxyType {
+	fn filter(&self, c: &Call) -> bool {
+		match self {
+			ProxyType::Any => true,
+			ProxyType::NonTransfer => matches!(
```
Comment (xlc) [inline runtime/mandala/src/lib.rs:1271]: make CancelProxy the second one to match the type defined for Karura

**Review by xlc — DISMISSED** [review]

**Review by shaunxw — DISMISSED** [review]

**Review by xlc — APPROVED** [review]

**Review by shaunxw — APPROVED** [review]

**Pull request merged**